### PR TITLE
Refactor/docs

### LINKED ISSUE #482: My proposal for refactoring the documentation
Congratulations on the stable release, and thank you so much for the amazing job you've done!

I'm an organization and documentation nerd 🤓, so I'd like to share my proposal to make the Hocuspocus documentation more clear.

The structure would be like this:
```
Overview
  Introduction
  Getting Started
  About the Project
  Contributing

Server
  Settings
  Methods
  Hooks*
  Extensions*
  Examples*

Provider
  Introduction
  Settings
  Events
  Examples*

Guides
  Authorization
  Persistence
  Multi/Sub-Documents
  Scalability
  Awareness
  Custom Extensions
```

Next I explain in more detail the changes of my proposal (some of which are evident with the previous structure and others are not):

- Make the main sections (`Overview`, `Server`, `Provider` and `Guides`) collapsible.
- Move the `Quick Start` subheading in the `Introduction` sub-section to `Installation` (there's already something very similar there, almost duplicated) and rename `Installation` to `Getting Started`.
- The `Awareness` sub-section is almost duplicated in the `Use Case` and `Provider` sections. I would consolidate them into a single sub-section under the Guides category, and make sure that `Guides > Awareness` and `Provider > Events` reference each other via a hyperlink.
- The rest of the sub-sections under `Use Case` move to `Provider > Examples`.
- Move the current sub-sub-sections under `Overview > installation` (Express, Koa, CLI, etc) to `Server > Examples`.
- Convert the `Server > Configuration`, `Server > Methods`, and `Provider > Configuration` sub-sections to tables instead of articles, as you already do with `Server > Hooks`. Due to the structured nature of the article, a table is much easier to scan than an article in such cases. The subtitle `Usage` at the end of those sections is nice and I would leave it that way.
- Delete the `Provider > Installation` section. The `Server` section doesn't have his, and the installation instructions are already mentioned in `Overview`. A hyperlink could be added at the top of `Provider > Introduction` to `Overview > Getting Started` for installation instructions.
- In the `Guides`, I would remove `Configuration` and `Hooks` as they are duplicate in the `Server` section. The `Documents`, `Working with JSON` and `Subdocuments` sections, I would refactor to two different guides: `Persistence` and `Multi/Sub-Documents`
- As you can see, the new structure that I propose is 2 levels deep, instead of 3. That is because the sections that I have marked with an asterisk have a table as content, with additional explanatory subheadings. If you're worried that people won't find that content because it's not in the sidebar, keep in mind that Algolia identifies the titles, and people will likely find the relevant links in the `Guides` section to the `Server` or `Provider` sections. ".

In my opinion that would be much clearer :)

I think if you like these suggestions I can collaborate with a PR. I see that the documentation consists of .md files with a .yml defining the structure. The only thing I'm not sure how to do is (1) make the main sections collapsible, and (2) run the documentation locally to see and debug the changes.

### PR DESCRIPTION
**Pull request by EgonBolton**
Resolve #482 

Well, it has taken a long time. Still room for improvement, but docs like this are much better, IMO.

Pending:

- [ ] Make the sections collapsible.
- [x] Add the Algolia search engine.
- [x] Please check that the internal links are correct. Since I have not been able to run them locally, I don't know if they are generated with the `links.yaml` file, if they are relative paths, etc. A lot of URLs would have to be changed, for example, `/api/server/extensions/database` to `/server/extensions#database`.
- [ ] I suggest adding the types to the different methods at some point. For some reason for example the documentation states that the `closeConnections` parameter is optional, but it does not do so with the `listen(...)` ones.

Doubts:
- In the server examples, especifically Express and Koa, I don't understand the following sentence: 

> some extensions use onRequest, onUpgrade and onListen hooks, which will not fire in this scenario. Please read the documentation of each extension to know how to use them when integrating Hocuspocus into the framework of your choice. 

I have left it as it is. But if you could explain me better what you are referring to maybe I can clarify that part.
- Another one a bit confusing, also in Express: 

> You have the option to use the `onConnect` hook of Hocuspocus or put your authorization and authentication code in your own update connection handler and pass the `context` as the fourth argument to the `handleConnection()` method.

Could you please clarify this part? I have some doubts. For example, why do you mention the `onConnect` hook instead of `onAuthenticate()`?
- There is an orphaned document called `transformations.md` that was part of the `Working with JSON` guide. If I'm not mistaken, that's what's listed in the Yjs documentation as bindings, right? https://docs.yjs.dev/ecosystem/editor-bindings 
I think it fits better there than as part of the Hocuspocus documentation, but correct me if I'm wrong.

### CODE REVIEW

**Comment by mkriegeskorte** [comment]
@EgonBolton Thanks for the PR. I need some time to review this, but I have it on my radar 🙌

**Comment by mkriegeskorte** [comment]
@EgonBolton Do you have a chance to resolve the conflicts? Guess you’ve a better overview of your desired changes :)

**Comment by EgonBolton** [comment]
I am not enabled for that.

image

If I'm not mistaken, all conflicts should be overwritten with the new changes.

**Comment by janthurau** [comment]
I'll rebase this.

**Comment by EgonBolton** [comment]
@mkriegeskorte @janthurau What would be missing to merge this PR?

File: docs/guides/persistence.md [hunk 1]
```diff
@@ -53,4 +51,6 @@ function createInitialDocTemplate() {
 
 ## FAQ: In what format should I save my document?
 
-In Uint8Array, [which is the format that Yjs encodes its documents](https://docs.yjs.dev/api/document-updates). You can persist your documents **ALSO** in another format like JSON if for some reason you want to. Other formats can give you the current state of the document, but you need your Y.Doc if you want the historical steps of the document, which is very much what RTC is all about!
+In Uint8Array, [which is the format that Yjs encodes its documents](https://docs.yjs.dev/api/document-updates).
+You can persist your documents **ALSO** in another format like JSON if for some reason you want to.
+Other formats can give you the current state of the document, but you need your Y.Doc. If you don't, you'll end up with duplicate content.
```
Comment (EgonBolton) [inline docs/guides/persistence.md:56]: > If you don't, you'll end up with duplicate content?
🤔

File: docs/links.yaml [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
     - title: Getting Started
-      link: /overview/getting-started
+      link: /getting-started
     - title: About the project
-      link: /overview/sponsor
+      link: /sponsor
       type: sponsor
     - title: Contributing
-      link: /overview/contributing
+      link: /contributing
```
Comment (EgonBolton) [inline docs/links.yaml]: I couldn't deploy, but is this ok?

**Review by janthurau — DISMISSED** [review]

**Pull request merged**